src/Build/Instance/ProjectItemInstance.cs

```diff
@@ -1096,40 +1093,45 @@ internal CopyOnWritePropertyDictionary<ProjectMetadataInstance> MetadataCollecti
 
                     CopyOnWritePropertyDictionary<ProjectMetadataInstance> allMetadata = new CopyOnWritePropertyDictionary<ProjectMetadataInstance>();
 
-                    // Next, any inherited item definitions. Front of the list is highest priority,
-                    // so walk backwards.
-                    for (int i = _itemDefinitions.Count - 1; i >= 0; i--)
+                    allMetadata.ImportProperties(metaData());
+
+                    return allMetadata;
+
+                    IEnumerable<ProjectMetadataInstance> metaData()
```

Comment:
Weird casing and naming.
GetOrderedMetadata/OrderMetadata/GetMetadata?

Reply:
I kinda like localFunction naming schema. Have seen it in Roslyn and use it since.

Reply:
I was rather refering to treating metadata as 2 words (metaData) and absence of verb in the function name.
First letter casing - my opinion - whatever is already used in our code... and if nothing - feel free to set the convention :-)

Reply:
These days I personally prefer not to include redundant information into naming. I tend to consider Get verb prefix as implicit. I mean if function name is just `nouns()` is is implicit that it will somehow GET me `nouns`. Given the small scope and close locality of this method, more describing names like `GetAllMetaDataOrderedAsLastWinForClonePurpose` is maybe not needed .
But It is matter of taste and I don't mind to change it though.